book/forms.md

```diff
@@ -13,349 +13,250 @@ out form information and then send that form to the server. Web forms
 are used almost everywhere: you fill one out to post on Facebook, to
 register to vote, or to search Google.
 
-::: {.todo}
-I may decide to implement focus tracking and GUI text entry.
-:::
-
 Rendering widgets
 =================
 
 When your browser sends information to a web server, that is usually
-information that you\'ve typed into some kind of input area, or a
-check-box of some sort that you\'ve checked. So the first step in
+information that you've typed into some kind of input area, or a
+check-box of some sort that you've checked. So the first step in
 communicating with other servers is going to be to draw input areas on
 the screen and then allow the user to fill them out.
 
 On the web, there are two kinds of input areas: `<input>` elements,
 which are for short, one-line inputs, and `<textarea>` elements, which
-are for long, multi-line text. I\'d like to implement both, because
-I\'d like to support both search boxes (where queries are short,
-single-line things) and comment forms (where text inputs are a lot
-longer).
-
-Both `<input>` and `<textarea>` elements are inline content, like
-text, laid out in lines. So to support inputs we\'ll need a new kind
-of layout object, which I\'ll call `InputLayout`. It\'ll need to
-support the same kind of API as `TextLayout`, namely `attach` and
-`add_space`, so that it won't confuse `InlineLayout`:
-
-``` {.python}
+are for long, multi-line text. I'll implement `<input>` only, because
+because `<textarea>` has a lot of strange properties.[^sig-ws]
+Usually, web browsers communicate with the operating system and ask
```

Comment:
Chrome does not do this..

Reply:
What's the right way to qualify this? "Many browsers..."? Maybe that's not even the right generalization here. Perhaps it should be

> Many applications want their input areas to match input areas for all other applications, so use operating-system libraries to draw an input area directly. That's possible in Tk, but browsers need a lot of control over application styling, so we'll be drawing input areas ourselves.

We can add a footnote saying that Safari and maybe Firefox do still sometimes use OS libraries for input areas, but need to detect when a web style makes that impossible.

Reply:
New text LGTM